packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -169,10 +175,39 @@ function coerceObject(input: string | object, spec: ParameterObject) {
   if (typeof data !== 'object' || Array.isArray(data))
     throw RestHttpErrors.invalidData(input, spec.name);
 
-  // TODO(bajtos) apply coercion based on properties defined by spec.schema
+  const schema = extractSchemaFromSpec(spec);
+  if (schema) {
+    // Apply coercion based on properties defined by spec.schema
+    const validate = AJV.compile(schema);
```

Comment:
See more details at https://github.com/strongloop/loopback-next/pull/1757. The `compile` call is probably the most expensive one.

Reply:
We probably have to maintain a weak map (schema -> ajv).